packages/runtime/test/transport/relationshipTemplates.test.ts

```diff
@@ -327,4 +327,54 @@ describe("RelationshipTemplates query", () => {
 
         await conditions.executeTests((c, q) => c.relationshipTemplates.getRelationshipTemplates({ query: q, ownerRestriction: OwnerRestriction.Peer }));
     });
+
+    test("password query for a template protected with a password that's not a PIN", async () => {
+        const passwordProtectedTemplate = await createTemplate(runtimeServices1.transport, undefined, { password: "password" });
+
+        const conditions = new QueryParamConditions<GetRelationshipTemplatesQuery>(passwordProtectedTemplate, runtimeServices1.transport).addStringSet(
+            "passwordProtection.password"
+        );
+        await conditions.executeTests((c, q) => c.relationshipTemplates.getRelationshipTemplates({ query: q }));
+    });
+
+    test("password query for a template protected with a PIN", async () => {
+        const pinProtectedTemplate = await createTemplate(runtimeServices1.transport, undefined, { password: "1234", passwordIsPin: true });
```

Comment:
actually it should be possible to add this test (and the other one) to the querying test above, just add the password protection and the filters.

Reply:
You can also add some completely custom queries and define if it should be matched or not.